core/dbt/parser/manifest.py

```diff
@@ -1629,6 +1630,19 @@ def invalid_target_fail_unless_test(
         )
 
 
+def warn_if_package_node_depends_on_root_project_node(
+    node: ManifestNode, target_model: ManifestNode, current_project: str
+) -> None:
+    if node.package_name != current_project and target_model.package_name == current_project:
```

Comment:
Do we want this to be slightly more targeted? My understanding is we're moving to a world where the package model will be preferred. That is the behavior flag, when flipped to true, will cause a matching model in the package to be referenced over the root project model IFF there is a matching model in the package. Is it guaranteed that the package has an implementation of the referenced model? If not, do we want to add another gate here that the `node.package_name` package must have a matching node? 🤔

Reply:
> do we want to add another gate here that the node.package_name package must have a matching node? 

Hmmm, although I can't think of a use case where you'd want this, I also don't see any reason to restrict that given that core will already raise an error for problematic behavior like cycles or unresolved refs.

Reply:
And specifically I'm thinking about the validity of the warning in different scenarios